text/0020-contributing-guidelines.md

```diff
@@ -0,0 +1,46 @@
+
+For frontend projects within OpenMRS, the following guidelines apply:
+- Not all PRs have to link to a JIRA ticket.
+- Pull requests are allowed (and encouraged) to contain more than one commit. Those commits are squashed when merged into the master branch.
+- Any pull request template within the repo should be followed.
+
+## Definition
+In this PR, contribution refers to people creating pull requests to Github repos. Other terms in the definition section have been linked to above.
+
+## Reason for decision
+- Github contributing.md and pull_request_template.md markdown files are the worldwide standard for open source software. They may link to a separate guide, but should be present.
+- The OpenMRS code of conduct helps provide a safe environment for people within the OpenMRS community.
+- The OpenMRS MPL-2.0 + Health care disclaimer software license protects the freedom of the open source software while also protecting OpenMRS from lawsuits.
+- The OpenMRS pull request tips are a community maintained set of good practices.
+- Github squash and merge creates the most easily read and understood commit history on the master branch.
+- Enforcing that every PR links to a JIRA ticket discourages contribution by increasing barrier to entry. First, you have to create a JIRA account. Then you have to create an issue in the right project. Then you have to self assign it. Then you have to link to it. Then you have to close it. Linking to JIRA is appropriate and helpful when you are working on a ticket. However, not every code change is described by a JIRA ticket. See #19 for more details on the role of JIRA.
+- Enforcing that every PR has only one commit increases the barrier to entry while destroying git history. It makes it harder for the reviewer to see incremental progress in a PR. Due to Github's squash and merge feature, it is not required for the master branch in order to have a clean commit history.
+
+## Alternatives
+The OpenMRS community up already adopts much of what is in this PR. The new aspects of it are using Github squash and merge, not enforcing that all PRs are associated with a JIRA ticket, and allowing for multiple commits in a single PR. These new aspects only apply to frontend code.
+
+## Common practices (not enforced)
+- Do not increase the scope of a PR after it has been reviewed.
+- Encourage feedback as both an author and reviewer.
+- Do not close a PR and recreate a new one with the same code.
+- Encourage as many contributors to engage in deep and meaningful contribution.
+- Be sensitive to the varying experience levels of OpenMRS contributors.
+- Avoid gatekeeping, which includes over-emphasizing to contributors how they're failing to follow the correct process.
+- Encourage code ownership. Review changes to the code you wrote. Do not overstep in a foreign codebase.
```

Comment:
Does the above mean that one does not review changes to code they did not write?

Reply:
No it does not mean that. The idea I'm communicating here is that it's good when people who understand the original code are the ones reviewing changes to it.

You are an example of someone who has taken a level of ownership over openmrs-core and a lot of repos -- your review is very valuable there and I think a good open source community encourages ownership like that over every repo. The webpack team for example, breaks up the repos among their team. They have sokra maintaining webpack itself, others maintaining webpack-dev-server, and others maintaining the webpack.js.org github repo. Each of the maintainers feels ownership over their area within the overall webpack project. It doesn't mean they can't code review PRs from other repos. Just means that they have an area that they're focused on.

Reply:
How about changing the language a bit, such that it does not scare away potential reviewers?

Reply:
👍 done - I removed that last sentence that was sort of confusing